### PR TITLE
Add test for kubectl support of ReplicationControllerList and ServiceList

### LINKED ISSUE #14775: Integration/e2e tests of kubectl support for multiple resources
Forked from #13812.

We advocate that users should specify multiple resources in a single file, yet we don't have adequate testing of that feature, and we broke it when we turned on schema validation by default.

It sounds like a test in test-cmd.sh would be adequate to exercise the default configuration of kubectl.

There are multiple ways to specify multiple resources:
* explicit List
* YAML document separator `---`
* JSON object concatenation(?)

Apparently different input sources are handled differently, at least for alternatives to an explicit List, so we probably need to at least test file and stdin. Not sure about directory and URL.

cc @liggitt @deads2k @feihujiang @smarterclayton @lavalamp

### PR DESCRIPTION
**Pull request by janetkuo**
Fixes #14775
Follow up https://github.com/kubernetes/kubernetes/pull/14793#issuecomment-144593911. 
Add test for ReplicationControllerList and ServiceList.

cc @smarterclayton @feihujiang @bgrant0607

### CODE REVIEW

**Comment by feihujiang** [comment]
We could remove the kind and apiVersion in each of items，then test

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -749,18 +751,45 @@ __EOF__
       replace_file="${file%.json}-modify.json"
     fi
 
-    ### Create, get, describe, replace, label, annotate, and then delete service nginxsvc and replication controller my-nginx from 3 types of files:
+    has_svc=true
+    has_rc=true
+    two_rcs=false
+    two_svcs=false
+    if [[ $file == *rclist* ]]; then
```
Comment (smarterclayton) [inline hack/test-cmd.sh]: nit: "${file}"

**Comment by smarterclayton** [comment]
Nits on bash - otherwise LGTM.

**Comment by JanetKuo** [comment]
Fixed nits. PTAL.

**Comment by JanetKuo** [comment]
@k8s-bot test this

**Comment by JanetKuo** [comment]
@k8s-bot test this

**Comment by wojtek-t** [comment]
@JanetKuo - please rebase this PR

**Comment by JanetKuo** [comment]
Rebased and squashed commits.

**Pull request merged**